test/e2e_node/critical_pod_test.go

```diff
@@ -85,6 +87,7 @@ var _ = SIGDescribe("CriticalPod [Serial] [Disruptive] [NodeFeature:CriticalPod]
 			// Create pods, starting with non-critical so that the critical preempts the other pods.
 			f.PodClient().CreateBatch([]*v1.Pod{nonCriticalBestEffort, nonCriticalBurstable, nonCriticalGuaranteed})
 			f.PodClientNS(kubeapi.NamespaceSystem).CreateSync(criticalPod)
+			e2epodclient.MakePodClientNS(f, kubeapi.NamespaceSystem).CreateSync(criticalPod)
```

Comment:
not sure why you add twice and then remove in a separate commit. The PR will merge as a whole, so it should be OK, but still feels weird.

Also, please note this test is broken, and we are working to fix it in https://github.com/kubernetes/kubernetes/pull/103408

Reply:
thanks for reviewing, this PR is old and many things might have changed since I created it, I think I wrote a document about all the planned changes linked in the description but I think I need to revisit this. I can squash all these changes too if needed.